[ZT712002] iP

ZT712002 opened the pull request.

ZT712002 committed:
Added Joebot

Start of Level-0 and project. Changed the text to the requirements of Level-0

ZT712002 committed:
Added Extra Bot Prompt Flavors

ZT712002 committed:
Finished Level 2 exercises, added list and inserting items function

ZT712002 committed:
fixed minor output bug where insertItem does not insert the first word into the array

ZT712002 committed:
Refactored code to comply with OOP standards

Added 2 new classes Task and TaskList respectively to comply with OOP
Standards. Tidied up main code as well.

ZT712002 committed:
Refactor Main

Changed "tasklist" to tasks to comply with A-CodingStandard for
submisison.

ZT712002 committed:
Add 2 class files

ZT712002 committed:
Add deadline, event, passer, todo

For Level-4, needed to do the required - deadline -event and todo
Created 3 new class files that inherits from task class to follow the
Inheritance requirement A-inheritance. Added Passer as a new class so
The main class old functionality is abstracted away to this Passer class

ZT712002 committed:
Fix bye command in Passer

Added Bye command when handling passer. Now program can exit the
Loop elegantly without throwing an exception error

ZT712002 committed:
Add test files

Added input and expected text files to fulfil text-ui-testing. Also ran
a linter to lint the rest of the class files.

ZT712002 committed:
Rename ToDo to Todo

ZT712002 committed:
Rename getUserinput to getUserInput

ZT712002 committed:
Rename in to IN

ZT712002 committed:
Change variable types

For classes Deadline, Events, Todo and Task removed final from their
variable types as I've realised in the future some of this variables
might be modified

Sheeeesh-code reviewed (commented):
Good Job, you followed the coding standard for the switch

Sheeeesh-code commented on src/main/java/Passer.java:
```diff
@@ -0,0 +1,111 @@
+        String command = commandParts[0].toLowerCase();
+        String arguments = commandParts.length > 1 ? commandParts[1] : "";
+        try {
+            switch (command) {
```
Good Job, The switch follows the java coding standard

Sheeeesh-code commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
```
All Class should be in a packages

ZT712002 replied:
Thanks for pointing this out! didn't realise that

Sheeeesh-code commented on src/main/java/Joebot666.java:
```diff
@@ -0,0 +1,17 @@
+public class Joebot666 {
```
Good job, you followed OOP, by making many Class for different utility, using inheritance

Sheeeesh-code commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+    private boolean isDone;
+
+
+    public Task(String description) {
```
Good job, your loop are  wrapped by curlys bracket even if there is only one line in the body

Sheeeesh-code commented on src/main/java/Passer.java:
```diff
@@ -0,0 +1,111 @@
+                break;
+            }
+
+        } catch (Exception e) {
```
Good job, your try catch statement have the good form

Sheeeesh-code commented on src/main/java/Passer.java:
```diff
@@ -0,0 +1,111 @@
+import java.util.Scanner;
+
+public class Passer {
+    private final Scanner IN;
```
Good job here IN is a constant hand you write it in uppercase

Sheeeesh-code reviewed (commented):
Good job, but don't forget to put all your Class in package

gordonajajar commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.ArrayList;
+
+public class TaskList {
```
I liked that you even extracted the Tasks array into its own class, not to mention the use of ArrayList for dynamic-length arrays.

gordonajajar commented on src/main/java/Passer.java:
```diff
@@ -0,0 +1,111 @@
+        return isPasserActive;
+    }
+
+    public void setPasserActive(boolean passerActive) {
```
I'm not sure setPasserActive is a descriptive name here, since it suggests that it will set Passer to active, but it in fact takes in a boolean and sets Passer to that value. Perhaps it can be factored into 2 different functions, setPasserActive and setPasserInactive? Or just something like setPasserState

ZT712002 replied:
Thanks, I've realised the only utility of this function would be to exit the program elegantly. I will refactor this in future commits. Thanks for pointing this out!

gordonajajar commented on src/main/java/Passer.java:
```diff
@@ -0,0 +1,111 @@
+
+    public void processCommand(String userInput, TaskList tasks) {
+        String[] commandParts = userInput.split(" ", 2);
+        String command = commandParts[0].toLowerCase();
```
Nice job pre-processing the input, even considering case

gordonajajar commented on src/main/java/Passer.java:
```diff
@@ -0,0 +1,111 @@
+    private void handleEvent(TaskList tasks, String arguments) {
+        String[] parts = arguments.split(" /from");
+        if (parts.length < 2) {
+            System.out.println("Please provide both description and start time for the event in this format " +
```
Indenting here is off; may be fixed by setting "Autoformat on save" in IntelliJ Idea? (If that is what you are using)

ZT712002 replied:
Thanks for catching that. Didn't realise :)

gordonajajar reviewed (commented):
Nice job! You have done a lot of abstraction as seen from your short Joebot666 (main) file.

ZT712002 committed:
Change setPasserActive functionality

setPassActive is now changed to setIsPasserActive off
reason for this is to just turn it off, by default is turned off so to
clear any confusion

ZT712002 committed:
Change Task class type

Task class is now abstract as all subclasses are called respectively and
task should not be created in the program

ZT712002 committed:
Add package directories

Added package directories to divide the classes into suitable java
packages to keep a clean root structure of the project. Also added the
import statements appropriately

ZT712002 committed:
Add error check

Added error check for keywords not detected by joebot

ZT712002 committed:
Merge branch 'branch-A-Packages' into branch-Level-5

ZT712002 committed:
Add exception handling

Created an IllegalTaskNumberException class that handles the error
checker between task numbers

ZT712002 committed:
Merge branch 'branch-Level-5'
Merge branch-Level-5 to Master

Merging as exception handling task
required for Level-5 is completed

Daviancold commented on src/main/java/Joebot666.java:
```diff
@@ -0,0 +1,20 @@
+        while (passer.isPasserActive()) {
+            System.out.println("*************************************");
+            passer.setUserInput();
+            passer.processCommand(passer.getUserInput(), tasks);
```
Just a small nitpick, but perhaps you could consider storing `passer.getUserInput()` into a variable before passing it into `processCommand()`

Daviancold commented on src/main/java/exception/IllegalTaskNumberException.java:
```diff
@@ -0,0 +1,4 @@
+package exception;
+
+public class IllegalTaskNumberException extends Exception {
```
Good and clear naming used for custom exception

ZT712002 committed:
Add delete method

Added delete method to delete tasks from list, Since my TaskList class
is already implemented with arraylist from the collections class, the
additional tag can be added as well.

ZT712002 committed:
Add check if path exists

ZT712002 committed:
Add check if file exists

ZT712002 committed:
Add save data method

ZT712002 committed:
Add load data method

Load data method now handles the text file elegantly and closes without
issues

ZT712002 committed:
Merge branch 'branch-Level-6'

ZT712002 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/command/Passer.java
#	src/main/java/logic/TaskList.java

ZT712002 committed:
Add import statement

ZT712002 committed:
Add jar manifest

ZT712002 committed:
Rename Passer to Ui

ZT712002 committed:
Add Storage Class

ZT712002 committed:
Add Command and Parser Class

ZT712002 committed:
Refactor Classes

ZT712002 committed:
Add date time format

ZT712002 committed:
Add event and deadline functionality

ZT712002 committed:
Merge pull request #1 from ZT712002/branch-Level-8

Add date and time formatting to save/store data

ZT712002 committed:
Add Find Function

ZT712002 committed:
Fix typo

ZT712002 committed:
Merge pull request #2 from ZT712002/branch-Level-9

Level 9 functionality

ZT712002 committed:
Add JavaDoc Comments

ZT712002 committed:
Merge pull request #3 from ZT712002/branch-A-JavaDoc

JavaDoc Documentation

ZT712002 committed:
Add User Guide

ZT712002 committed:
Add table of contents

ZT712002 committed:
Add Date error Check

ZT712002 committed:
Update root readme

ZT712002 committed:
Fix readme typo

ZT712002 committed:
Fix deadline bug

ZT712002 committed:
Fix repeat welcome message bug

ZT712002 committed:
Fix line divider bug

ZT712002 committed:
Add more JavaDoc Comments

ZT712002 committed:
Add newlines for end of java class
